File: 922/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(sortArrayByParityII([]int{4,2,5,7}))
	fmt.Println(sortArrayByParityII([]int{4,3}))
}

func sortArrayByParityII(nums []int) []int {
	ans := make([]int, len(nums))
	s := []int{}
	d := []int{}
	for i := 0; i < len(nums); i++ {
		if nums[i] % 2 == 0 {
			s = append(s, nums[i])
		} else {
			d = append(d, nums[i])
		}
	}
	si := 0
	di := 0
	for i := 0; i < len(nums); i++ {
		if i % 2 == 0 {
			ans[i] = s[si]
			si++
		} else {
			ans[i] = d[di]
			di++
		}
	}
	return ans
}

File: 922/main.rs
struct Solution {}

impl Solution {
    pub fn sort_array_by_parity_ii(nums: Vec<i32>) -> Vec<i32> {
        let mut ans = vec![0; nums.len()];
        let mut i = 0;
        for &n in &nums {
            if n % 2 == 0 {
                ans[i] = n;
                i += 2;
            }
        }
        i = 1;
        for &n in &nums {
            if n % 2 == 1 {
                ans[i] = n;
                i += 2;
            }
        }
        return ans;
    }
}

fn main() {
    println!("{:#?}", Solution::sort_array_by_parity_ii(vec![4,2,5,7]));
}